ios general/ios general/Controller/MainTableViewController.m

```diff
@@ -0,0 +1,129 @@
+#import "MainTableViewController.h"
+#import "DetailViewController.h"
+#import "CreateEmployeeViewController.h"
+#import "EmployeeMO+Custom.h"
+#import "OrganizationMO+CoreDataClass.h"
+#import "AppDelegate.h"
+
+@interface MainTableViewController () <AddControllerDelegate>
+
+@property (nonatomic, strong) NSArray<EmployeeMO *> *employees;
+@property (nonatomic, weak) EmployeeMO *selectedEmployee;
+@property (nonatomic, strong) OrganizationMO *currentOrganization;
+
+@end
+
+@implementation MainTableViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    self.tableView.allowsMultipleSelectionDuringEditing = NO;
+    // Check if Organization exist
+    NSManagedObjectContext *context = [AppDelegate shared].managedObjectContext;
+    NSError *error = nil;
+    NSArray *results = [context executeFetchRequest:[OrganizationMO fetchRequest] error:&error];
+    
+    if (!results || [results count] == 0)
+    {
+        // if Organization doesn't exist, create one
+        self.currentOrganization = [NSEntityDescription insertNewObjectForEntityForName:@"Organization" inManagedObjectContext:context];
+        self.currentOrganization.name = @"Default Organization";
+    }
+    else
+    {
+        self.currentOrganization = [results objectAtIndex:0];
+    }
+    
+    self.employees = [self sortEmployeeSetByKey:self.currentOrganization.employees key:@"firstName"];
```

Comment:
Copy-pasted logic for sorting. Create a read-only property in Organization extension.